File: crop.py
from PIL import Image

# Open the image
img = Image.open('logo.png')

# Convert to RGBA if not already
img = img.convert("RGBA")

# Get bounding box of non-transparent pixels
bbox = img.getbbox()

if bbox:
    # Crop the image to the bounding box
    img_cropped = img.crop(bbox)
    # Save the new image, with a new filename to bust browser cache
    img_cropped.save('frontend-web/public/logo-cropped.png')
    print("Successfully cropped and saved to logo-cropped.png")
else:
    print("Image is entirely transparent")

File: crop.js
const Jimp = require("jimp");

Jimp.read("logo.png")
  .then(image => {
    // Autocrop with a tolerance of 5% (0.05) to catch JPEG compression artifacts in white padding
    return image.autocrop(0.05, false).writeAsync("frontend-web/public/logo-v4.png");
  })
  .then(() => {
    console.log("Cropped successfully to frontend-web/public/logo-v4.png");
  })
  .catch(err => {
    console.error("Error cropping image:", err);
  });
